homework3/semi-uber/Server/api/routes/user.router.js

```diff
@@ -0,0 +1,28 @@
+const express = require("express");
+const router = express.Router();
+const config = require("config");
+const log4js = require("log4js");
+const logger = log4js.getLogger();
+const { User } = require("../Schemas/user.schema");
+const writeLog = require("../middlewars/writeLog");
+const jwt = require("jsonwebtoken");
+const { secret } = config.get("JWT");
+
+router.get("/:id", writeLog, (req, res) => {
+  const pageID = req.params.id;
+  const token = req.headers["token"];
+  const userID = jwt.verify(token, secret);
+  if (pageID !== userID) {
+    logger.error("Access rejected");
+    res.status(403).json({ status: "Access rejected" });
+    throw new Error();
```

Comment:
It is not neccecary to throw an error here, you have already responded with an error to your server "knows" that somwthing went wrong and it was already handeled